### ISSUE TITLE
Is Http3.Roundtripper working for non-QUIC connections?

### ISSUE THREAD

**Issue by klaasel** [lucas-clemente/quic-go#1996]
I successfully implemented Quic-go in a Go server on a Windows VM and on a React Native Android app which normally uses a Go HttpClient (the React Native platform implements a bridge to Go). FYI: see https://github.com/privacybydesign/. The app successfully connects to the server, in WireShark you will see a lot of UDP messages. 

On the Android app I assigned the &http3.Roundtripper{} as documented. However if I want to connect to a different (non-QUIC) server within the app, the connection does time-out. Is this by design or should the http3.Roundtripper work as well with the regular TCP connection as fallback?

**Comment by marten-seemann** [comment]
HTTP/3 only works on top of QUIC.

**Comment by lucas-clemente** [comment]
I think this is a valid point, and we should at least update the documentation, and probably add a new roundtripper that does what users will expect.

As Marten describes, the http3.Roundtripper right now only supports HTTP3. That makes sense on the surface, but isn't very practical outside of a few edge case deployments. In practice, you probably want to default to H1/2, and then upgrade to H3 after discovering the Alt-Svc header. Alternatively, we could implement racing logic, but that is likely much harder. The main point is that there should be a way to support H3 whenever it makes sense, but without regressing H1/2.

Let's update the docs, and keep this issue open as a FR for a (wrapper around the) roundtripper that supports H1-3.

**Comment by mholt** [comment]
Yes, that sounds like it would be very helpful! I'm prepared to offer Caddy 2 as a testbed for the IETF HTTP/3 Alt-Svc-aware round-tripper implementation.

**Comment by klaasel** [comment]
For my interest: is the fallback to HTTP which the server does have specified in the IETF drafts?

**Comment by dropwhile** [comment]
Is there a recommendation for client implementations wrt quic-go either upgrading from or falling back to http/2,1.1 with http3.RoundTripper?

**Comment by codenoid** [comment]
I think the implementation must be on higher layer like https://github.com/go-resty/resty wrapper, but yeah, It will be such a bliss if we can fallback to HTTP/2 or 1
